### PR TITLE
Add TextureMetadata::srgb_scale for defaults with nearest filter

### PR DESCRIPTION
**Pull request by NCrashed**
Linear interpolation filter (current new default for `TextureMetadata::srgb()`) plays bad with sprites. 2D setups usually use nearest filter. The PR restores old behavior for sprites examples.

* Added new function `TextureMetadata::srgbScale()`
* Pong tutorial is adjusted to use nearest filter 
* Sprites examples are adjusted to use nearest filter 
* Added note in book when to use srgbScale

How to check:
* Check examples pong_tutorial_*, sprites, sprites_ordered
* Check docs

### CODE REVIEW

File: amethyst_renderer/src/formats/texture.rs [hunk 1]
```diff
@@ -70,6 +70,15 @@ impl TextureMetadata {
         }
     }
 
+    /// Creates texture metadata for `Srgb` texture but without texture filtering. This is usually
+    /// the case for color textures for 2D sprites and pixel art textures.
+    ///
+    /// For the values of all the other fields please refer to the documentation of the respective
+    /// field.
+    pub fn srgbScale() -> Self {
```
Comment (Moxinilian) [inline amethyst_renderer/src/formats/texture.rs]: You must use snake_case for functions, according to Rust's formatting guidelines.
`srgb_nearest` maybe?

File: book/src/sprites/load_the_texture.md [hunk 2]
```diff
@@ -53,6 +53,8 @@ There are two things that may surprise you.
 
     You pick the texture ID based on how you want to reference it. For example, you might have an application configuration that says `path/to/spritesheet_0.png` is ID `100`, `path/to/spritesheet_1.png` is ID `101`, so you can use that. Or, you might do something clever like calculate an ID based on the path, and if it's already loaded, then you know you don't have to load it again.
 
+The loaded texture will use linear filter, e.g. screen pixels will be linearly interpolated between the closest image pixels. If your images have small resolution, sprites will look blury. Use `TextureMetadata::srgbScale()` instead to avoid such effect. Screen pixel will be taken from nearest pixel of texture in that case.
```
Comment (Moxinilian) [inline book/src/sprites/load_the_texture.md]: "[...] image pixels. In layman's terms, if your images have [...]"

**Comment by NCrashed** [comment]
Your welcome, I am trying to help to make the engine more awesome

File: book/src/sprites/load_the_texture.md [hunk 3]
```diff
@@ -53,6 +53,8 @@ There are two things that may surprise you.
 
     You pick the texture ID based on how you want to reference it. For example, you might have an application configuration that says `path/to/spritesheet_0.png` is ID `100`, `path/to/spritesheet_1.png` is ID `101`, so you can use that. Or, you might do something clever like calculate an ID based on the path, and if it's already loaded, then you know you don't have to load it again.
 
+The loaded texture will use linear filter, e.g. screen pixels will be linearly interpolated between the closest image pixels. In layman's terms, i your images have small resolution, sprites will look blury. Use `TextureMetadata::srgb_scale()` instead to avoid such effect. Screen pixel will be taken from nearest pixel of texture in that case.
```
Comment (jojolepro) [inline book/src/sprites/load_the_texture.md]: "i your images"

Comment (NCrashed) [reply book/src/sprites/load_the_texture.md]: Ops, thanks, will fix now

**Comment by NCrashed** [comment]
Squashed

File: amethyst_renderer/src/formats/texture.rs [hunk 4]
```diff
@@ -70,6 +70,15 @@ impl TextureMetadata {
         }
     }
 
+    /// Creates texture metadata for `Srgb` texture but without texture filtering. This is usually
+    /// the case for color textures for 2D sprites and pixel art textures.
+    ///
+    /// For the values of all the other fields please refer to the documentation of the respective
+    /// field.
+    pub fn srgb_scale() -> Self {
```
Comment (torkleyy) [inline amethyst_renderer/src/formats/texture.rs]: The wrap mode should either be taken as parameter or clearly documented.

**Comment by Rhuagh** [comment]
Changelog conflict, ready to push after that is fixed.

**Comment by Rhuagh** [comment]
Ah, and fix @torkleyy's comment.

**Comment by Rhuagh** [comment]
There are lint errors, please fix.

**Comment by Rhuagh** [comment]
Bors r+
